dashboards/dash_all_metrics/modules/v2/perp_integrators.py

```diff
@@ -0,0 +1,213 @@
+        WHERE ts >= '{start_date}' AND ts <= '{end_date}'
+        ORDER BY ts
+        """
+    )
+
+    return {
+        "integrator_stats_agg": df_integrator_stats_agg,
+    }
+
+
+@st.cache_data(ttl="30m")
+def make_charts(data):
+    """
+    Creates charts based on the fetched data.
+
+    Args:
+        data (dict): A dictionary containing fetched dataframes.
+
+    Returns:
+        dict: A dictionary containing Plotly chart objects.
+    """
+    df = data["integrator_stats_agg"]
+    return {
+        "volume": chart_many_bars(
+            df,
+            "ts",
+            ["volume"],
+            "Volume",
+            "tracking_code",
+        ),
+        "volume_pct": chart_many_bars(
+            df,
+            "ts",
+            ["volume_share"],
+            "Volume %",
+            "tracking_code",
+            y_format="%",
+        ),
+        "exchange_fees": chart_many_bars(
+            df,
+            "ts",
+            ["exchange_fees"],
+            "Exchange Fees",
+            "tracking_code",
+        ),
+        "exchange_fees_pct": chart_many_bars(
+            df,
+            "ts",
+            ["exchange_fees_share"],
+            "Exchange Fees %",
+            "tracking_code",
+            y_format="%",
+        ),
+        "trades": chart_many_bars(
+            df,
+            "ts",
+            ["trades"],
+            "Trades",
+            "tracking_code",
+            y_format="#",
+        ),
+        "trades_pct": chart_many_bars(
+            df,
+            "ts",
+            ["trades_share"],
+            "Trades %",
+            "tracking_code",
+            y_format="%",
+        ),
+        "traders": chart_many_bars(
+            df,
+            "ts",
+            ["traders"],
+            "Traders",
+            "tracking_code",
+            y_format="#",
+        ),
+        "cumulative_volume": chart_many_lines(
+            df,
+            "ts",
+            ["cumulative_volume"],
+            "Cumulative Volume",
+            "tracking_code",
+        ),
+        "cumulative_exchange_fees": chart_many_lines(
+            df,
+            "ts",
+            ["cumulative_exchange_fees"],
+            "Cumulative Exchange Fees",
+            "tracking_code",
+        ),
+        "cumulative_trades": chart_many_lines(
+            df,
+            "ts",
+            ["cumulative_trades"],
+            "Cumulative Trades",
+            "tracking_code",
+        ),
+    }
+
+
+def main():
+    """
+    The main function that sets up the Streamlit dashboard.
+    """
+    # Initialize session state for filters if not already set
+    if "resolution" not in st.session_state:
+        st.session_state.resolution = "daily"
+    if "start_date" not in st.session_state:
+        st.session_state.start_date = datetime.today().date() - timedelta(days=14)
+    if "end_date" not in st.session_state:
+        st.session_state.end_date = datetime.today().date()
```

Comment:
Though it's not related to this specific PR, I think we should try and streamline a bit the way we are using `resolution` in different places in the data pipeline (`daily`/`hourly` for table & view names vs `24h`/`7d`/`28d` for APR fields), as I think it can impact stuff in these tools.

Reply:
I noticed this won't get messed up since these get initialized by their components before the queries run. It's a good point though and we should just rename one.